tests/unit/connect-test.js

```diff
@@ -59,3 +59,35 @@ test('connect with remote being promise', function (t) {
 
   t.end()
 })
+
+test('connect replication error', function (t) {
+  t.plan(3)
+
+  var syncApi = {
+    on: simple.stub()
+  }
+  var state = {
+    emitter: {
+      emit: simple.stub()
+    },
+    remote: 'remoteDb',
+    db: {
+      sync: simple.stub()
+    }
+  }
+  simple.mock(state.db, 'sync').returnWith(syncApi)
+
+  connect(state)
+
+  .then(function () {
+    t.is(syncApi.on.calls[0].args[0], 'error', 'listens "error" event')
```

Comment:
gotcha, yes that’s correct. You could still use the shortcut `.arg` for `.args[0]` though, so `syncApi.on.calls[0].arg` in this case. Or just leave it as is, maybe it’s even more clear that way :)